airflow/models.py

```diff
@@ -3978,6 +3983,11 @@ def update_state(self, session=None):
         """
 
         dag = self.get_dag()
+
+        # backfills create subsets of dags with the same name
```

Comment:
I think the proper way of doing it, is not to create a subdag but let the backfill job (/ scheduler job in the future) figure this out properly. Ie. sub_dag should be renamed "subset" and we could just filter out what tasks to run from the whole dag.